modules/control-utility/src/test/java/org/apache/ignite/util/GridCommandHandlerCheckpointTest.java

```diff
@@ -0,0 +1,278 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.util;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.regex.Pattern;
+import org.apache.ignite.IgniteCache;
+import org.apache.ignite.cluster.ClusterState;
+import org.apache.ignite.configuration.DataRegionConfiguration;
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.processors.cache.persistence.GridCacheDatabaseSharedManager;
+import org.apache.ignite.internal.processors.cache.persistence.checkpoint.CheckpointListener;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.apache.ignite.testframework.ListeningTestLogger;
+import org.apache.ignite.testframework.LogListener;
+import org.junit.Test;
+
+import static org.apache.ignite.internal.commandline.CommandHandler.EXIT_CODE_OK;
+
+/** Test for checkpoint in control.sh command. */
+public class GridCommandHandlerCheckpointTest extends GridCommandHandlerAbstractTest {
+    /** */
+    private static final String PERSISTENT_REGION_NAME = "pds-reg";
+
+    /** */
+    private final ListeningTestLogger listeningLog = new ListeningTestLogger(log);
+
+    /** */
+    private final LogListener checkpointFinishedLsnr = LogListener.matches("Checkpoint finished").build();
+
+    /** */
+    private boolean mixedConfig;
+
+    /** Latch for blocking checkpoint in timeout test. */
+    private CountDownLatch blockCheckpointLatch;
+
+    /** */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        listeningLog.registerListener(checkpointFinishedLsnr);
+
+        cfg.setGridLogger(listeningLog);
+
+        if (mixedConfig) {
+            DataStorageConfiguration storageCfg = new DataStorageConfiguration();
+
+            storageCfg.setDefaultDataRegionConfiguration(new DataRegionConfiguration().setName("default_in_memory_region")
+                                                                                      .setPersistenceEnabled(false));
+
+            if (igniteInstanceName.contains("persistent_instance")) {
+                DataRegionConfiguration persistentRegionCfg = new DataRegionConfiguration();
+
+                storageCfg.setDataRegionConfigurations(persistentRegionCfg.setName(PERSISTENT_REGION_NAME)
+                                                                          .setPersistenceEnabled(true));
+            }
+
+            cfg.setDataStorageConfiguration(storageCfg);
+        }
+        else if (!persistenceEnable()) {
+            cfg.setDataStorageConfiguration(null);
+        }
```

Comment:
Also, we don't use braces for one-line "if" statements, see Ignite [codestyle](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)